### ISSUE TITLE
Please add BPMN 2 support

### ISSUE THREAD

**Issue by zhichkin** [elsa-workflows/elsa-core#305]
Your project is awesome ! Do you have any plans to add support for BPMN 2 ?
Do you have documentation on Elsa's JSON format used to serialize workflow diagrams ? I'm thinking on implementing some transformation tool from BPMN 2 to Elsa's format ...

**Comment by sfmskywalker** [comment]
Thanks! :)

I don't have any concrete plans as of yet due to other priorities with the project, but I definitely do intend to research the possibilities at some point in the future. If someone wants to do it sooner, then I am all for it.

I don't have documentation on the JSON format, but it's a good idea to do that. Right now it;s still in flux, especially with the work being done on Elsa 2. But the basic structure is as follows (example):

```json
{
  "activities": [
    {
      "name": "WriteLine",
      "id": "activity-1",
      "textExpression": {
        "syntax": "PlainText",
        "expression": "Hi! What's your name?"
      }
    },
    {
      "id": "activity-2",
      "name": "ReadLine",
      "argumentName": "name"
    },
    {
      "name": "WriteLine",
      "id": "activity-3",
      "textExpression": {
        "syntax": "JavaScript",
        "expression": "`Nice to meet you, ${name}!`"
      }
    }
  ],
  "connections": [
    {
      "source": {
        "activityId": "activity-1",
        "name": "Done"
      },
      "target": {
        "activityId": "activity-2"
      }
    },
    {
      "source": {
        "activityId": "activity-2",
        "name": "Done"
      },
      "target": {
        "activityId": "activity-3"
      }
    }
  ]
}
```

So basically a root node containing two fields: `Activities` and `Connections` that holds connection information between two activity nodes.

I'm closing this issue as a duplicate of [this one](https://github.com/elsa-workflows/elsa-core/issues/39). Let's continue the conversation there!
